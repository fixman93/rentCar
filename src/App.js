import React, { Component } from 'react';
import { Switch, Route } from 'react-router-dom'
import { library } from '@fortawesome/fontawesome-svg-core'

import { faPhone, faClock, faCheck, faCamera } from '@fortawesome/free-solid-svg-icons'
import NavBar from './components/navbar'
import Footer from './components/footer'
import Dashboard from './components/dashboard/index'
import CreateProjects from './components/createProject/createProject'
import Login from './components/login/index'
import Profile from './components/profile/index'
import ProjectDetail from './components/carDetail/ProjectDetail'
import Home from './components/home/index'
import Search from './components/home/Search/Search'
import './App.scss';

library.add(faPhone)
library.add(faClock)
library.add(faCheck)
library.add(faCamera)

class App extends Component {
  render() {
    return (
      <div>
        <NavBar />
        <Switch>
          <Route exact path='/' component={Home} />
          <Route path='/project/:id' component={ProjectDetail} />
          <Route path='/dashboard' component={Dashboard} />
          <Route path='/login' component={Login} />
          <Route path='/profile' component={Profile} />
          <Route path='/add-new-car' component={CreateProjects} />
          <Route path='/search/:id' component={Search} />
        </Switch>
        <Footer />
      </div>
    );
  }
}

export default App;
