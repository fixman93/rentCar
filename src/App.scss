body {
  font-family: 'Poppins', sans-serif;
}

a:hover {
  text-decoration: none;
}
.title-block {
  font-weight: 500;
  margin-bottom: 60px;
  color: #fff;
  font-size: 36px;
  line-height: 1;
}

button.btn.btn-primary {
  color: #fff;
  background-color: #247bdd;
  border-color: #247bdd;
  box-shadow: none;
  border-radius: 0;
  min-width: 200px;
  padding: 14px 10px;
  font-size: 20px;
  text-transform: uppercase;
}

.text-center {
  text-align: center;
}

.text-gray {
  color: #333;
}

.font-bold {
  font-weight: bold;
}

.divider-30 {
  margin: 30px 0;
}

.divider {
  font-size: 0;
  line-height: 0;
  height: 1px;
  width: 100%;
  background: #e1e1e1;
}

.form-group label {
  font-weight: normal;
  font-size: 13px;
  text-transform: none;
  color: #231f20;
  font-family: 'Roboto Mono', monospace;
  margin-bottom: 3px;
}

.form-group label {
  font-weight: normal;
  font-size: 13px;
  text-transform: none;
  color: #231f20;
  font-family: 'Roboto Mono', monospace;
  margin-bottom: 3px;
}

.form-control {
  height: 35px;
  padding: 3px 10px;
  font-size: 13px;
  border: 1px solid #c7c7c7;
  color: #231f20;
  width: 100%;
  -webkit-appearance: none;
  border-radius: 0;
}

.form-group .add-elements {
  padding: 0 10px;
  border-color: #c7c7c7;
}