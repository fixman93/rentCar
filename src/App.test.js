import React from 'react'
import Adapter from 'enzyme-adapter-react-16';

import { shallow, configure } from 'enzyme'

import App from './App'

configure({ adapter: new Adapter() });

describe('App', () => {
  const app = shallow(<App />)
  it('render propertly App', () => {
    expect(app).toMatchSnapshot()
  })

  it('contains a  NavBar component', () => {
    expect(app.find('Connect(NavBar)').exists()).toBe(true)
  })

  it('contains a Footer component', () => {
    expect(app.find('Footer').exists()).toBe(true)
  })
})