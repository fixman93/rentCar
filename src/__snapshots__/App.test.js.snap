// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render propertly App 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(NavBar) />,
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/project/:id"
          />
          <Route
            component={[Function]}
            path="/dashboard"
          />
          <Route
            component={[Function]}
            path="/login"
          />
          <Route
            component={[Function]}
            path="/profile"
          />
          <Route
            component={[Function]}
            path="/add-new-car"
          />
          <Route
            component={[Function]}
            path="/search/:id"
          />
        </Switch>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Route
              component={[Function]}
              path="/project/:id"
            />,
            <Route
              component={[Function]}
              path="/dashboard"
            />,
            <Route
              component={[Function]}
              path="/login"
            />,
            <Route
              component={[Function]}
              path="/profile"
            />,
            <Route
              component={[Function]}
              path="/add-new-car"
            />,
            <Route
              component={[Function]}
              path="/search/:id"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/project/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/dashboard",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/login",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/profile",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/add-new-car",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "path": "/search/:id",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(NavBar) />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              path="/project/:id"
            />
            <Route
              component={[Function]}
              path="/dashboard"
            />
            <Route
              component={[Function]}
              path="/login"
            />
            <Route
              component={[Function]}
              path="/profile"
            />
            <Route
              component={[Function]}
              path="/add-new-car"
            />
            <Route
              component={[Function]}
              path="/search/:id"
            />
          </Switch>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                path="/project/:id"
              />,
              <Route
                component={[Function]}
                path="/dashboard"
              />,
              <Route
                component={[Function]}
                path="/login"
              />,
              <Route
                component={[Function]}
                path="/profile"
              />,
              <Route
                component={[Function]}
                path="/add-new-car"
              />,
              <Route
                component={[Function]}
                path="/search/:id"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/project/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/dashboard",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/profile",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/add-new-car",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/search/:id",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
